src/main/java/com/github/manolo8/darkbot/core/objects/swf/SwfPtrCollection.java

```diff
@@ -1,27 +1,47 @@
 package com.github.manolo8.darkbot.core.objects.swf;
 
 import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.itf.UpdatableAuto;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Consumer;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 
 /**
  * Represents a collection of pointers in SWF.
  * Contains an utility method for syncing the pointers with java object collections.
  */
-public interface SwfPtrCollection {
+public abstract class SwfPtrCollection extends Updatable {
+    private long lastPointer;
+
     /**
      * @return size of the SWF collection
      */
-    int getSize();
+    public abstract int getSize();
 
     /**
      * @param i The index to search
      * @return The pointer the index points to to
      */
-    long getPtr(int i);
+    public abstract long getPtr(int i);
+
+    public void forEachIndexed(Consumer<Long> consumer) {
```

Comment:
This will need a better name, `forEachIndexed` does not sound like it will keep track of last position and continue from there